### ISSUE TITLE
Deprecate python switching functionality?

### ISSUE THREAD

**Issue by breznak** [joerick/cibuildwheel#365]
Unless it's needed for some other CIs, could we drop the python switching functionality in this package, and leave that to `actions/setup-python` (in GH Actions)?

**Comment by Czaki** [comment]
It is needed for other CI and increase reproducibility of builds.

**Comment by YannickJadoul** [comment]
What do you mean with "python switching functionality"?

**Comment by breznak** [comment]
> What do you mean with "python switching functionality"? 

selecting the python interpreter(s) to build the wheels with. 
```
# Only build on Python >=3.7 and skip 32-bit builds
CIBW_BUILD: cp37-*
CIBW_SKIP: "*-win32 cp27-manylinux* *i686*"
```
> And while we're at it: why would you want it to be dropped? 

because the functionality is already provided by another GH action: https://github.com/actions/setup-python 

> It is needed for other CI and increase reproducibility of builds.

this resolves it if the ability is needed/not provided-by other CI platforms.

**Comment by YannickJadoul** [comment]
The whole point of `cibuildwheel` is to do this all for you, running `cibuildwheel` only once and it taking care of everything else, installing and looping over valid Python versions, consistently over different platforms. On Linux, this loop is even inside a docker image, so the GitHub Action won't do anything useful.

This is (to me) the core of `cibuildwheel`, so ... (without speaking for the other maintainers) no. If you don't need all the fancy stuff around it, just want test on multiple Python versions and the GitHub Action works for you: go ahead. But as far as I can see, I respectfully think this is 1) not an issue and 2) definitely not a solution for `cibuildwheel`.

**Comment by breznak** [comment]
> But as far as I can see, I respectfully think this is 1) not an issue and 2) definitely not a solution for cibuildwheel. 

I understand your direction, this is then out-of scope for this project. Closing. 
Thanks for explaining.
